content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2353 @@
+data.
+
+The only currently defined schema for which custom values are used is -53
+(custom buckets). The remaining part of this section describes the usage of the
+custom values in more detail for this specific case.
+
+The custom values represent the upper inclusive boundaries of the custom
+buckets. They are sorted in ascending fashion. The custom buckets themselves
+are stored using the positive bucket list and the positive span list, although
+their boundaries, as determined via the custom values, can be negative. The
+index of each of those “positive” buckets defines the zero-based position of
+their upper boundary within the custom values list.
+
+The lower exclusive boundary is defined by the custom value preceding the upper
+boundary. For the first custom value (at position zero in the list), there is
+no preceding value, in which case the lower boundary is considered to be
+`-Inf`. Therefore, the custom bucket with index zero counts all observations
+between `-Inf` and the first custom value. In the common case that only
+positive observations are expected, the custom bucket with index zero SHOULD
+have an upper boundary of zero to clearly mark if there have been any
+observations at zero or below. (If there are indeed only positive observations,
+the custom bucket with index zero will stay unpopulated and therefore will
+never be represented explicitly. The only cost is the additional zero element
+at the beginning of the custom values list.)
+
+The last custom value MUST not be `+Inf`. Observations greater than the last
+custom value go into an overflow bucket with an upper boundary of `+Inf`. This
+overflow bucket is added with an index equal to the length of the custom
+values list.
+
+### Exemplars
+
+A native histogram sample can have zero, one, or more exemplars. They work in
+the same way as conventional exemplars, but they are organized in a list (as
+there can be more than one), and they MUST have a timestamp.
+
+Exemplars exposed as part of a classic histogram MAY be used by native
+histograms, if they have a timestamp.
+
+### Special cases of observed values
+
+Instrumented code SHOULD avoid observing values of `NaN` and `±Inf` because
+they make limited sense in the context of a histogram. However, those values
+MUST still be handled properly, as described in the following.
+
+The sum of observations is calculated as usual by adding the observation to the
+sum of observations, following normal floating point arithmetic. (For example,
+an observation of `NaN` will set the sum to `NaN`. An observation of `+Inf`
+will set the sum to `+Inf`, unless it is already `NaN` or `-Inf`, in which case
+the sum is set to `NaN`.)
+
+An observation of `NaN` goes into no bucket, but increments the count of
+observations. This implies that the count of observations can be greater than
+the sum of all buckets, and the difference is the number of `NaN` observations.
```

Comment:
Should it mention that for histograms that haven't had any NaN observations/have a sum of NaN, the count has to be equal to the sum of overall counts in the negative buckets, positive buckets, and zero count?